internal/manifest/commit_deployment.go

```diff
@@ -30,6 +31,9 @@ type OSTreeDeployment struct {
 	Keyboard            string
 	Locale              string
 
+	Users  []users.User
+	Groups []users.Group
```

Comment:
These should be IMO added to `OSCustomizations` instead.

Reply:
I considered it and flip flopped a bit while making this change.  There's a lot of things in `OSCustomizations` that we wouldn't use here and having them available might be misleading.  Perhaps we could try to cover as many as we can.

Some things about this new design aren't final yet (not sure if `OSCustomizations` is one of those things; cc @teg?) and I'm not sure we landed on a best approach so I have no strong feelings one way or another.

Reply:
Yeah, I was also not sure what was the real intention with `OSCustomizations`, but my previous ideas (also related to `ImageConfig`) were that there could be a master config structure for each particular image type, which would get passed to each pipeline and the pipeline would use the relevant parts from it. E.g. when you take the image installer image types. You can configure keyboard, timezone, etc. in the payload, as well as in the anaconda tree. My expectation was that we would like to ideally use the same config for all trees in that image, but I'm sure that someone may have a use case to not do that.

If not using `OSCustomizations`, we could at least standardize on `XXXCustomizations` structure for each pipeline. I'm not sure if we will ever end up defining images outside of the code (as a config file), but that would IMO make it easier to do. Right now, AFAICT we lost the ability to inherit customizations from other definition, which is something that we currently use for some RHEL images (e.g. RHUI vs. non-RHUI).

Reply:
I agree on standardising the structure.  Maybe we could have blocks of customisations that make sense together (language, keymap, timezone, and ntp; users and groups; packages, excludes, and repos; kernel stuff; things like that) and those can be combined into bigger structures in each image kind.